### ISSUE TITLE
Wrong overload resolution with a splat and named parameter

### ISSUE THREAD

**Issue by beta-ziliani** [crystal-lang/crystal#12596]
Covering test cases for #12584 I found that adding a parameter after the splat make overloading take the wrong definition:

```cr
def foo(*nums : UInt32, y : Int32)
  1
end

def foo(*nums : UInt32, y : Int32)
  2
end

foo(1, y: 1) # => 1
```

I'm not tagging this as a regression, since it wasn't working before either.

**Comment by jhass** [comment]
Technically an override, not an overload?

**Comment by beta-ziliani** [comment]
@jhass ah yes, but it's taken by the overload resolver :-)
@HertzDevil nice to know, definitively not worth the effort now then.
@straight-shoota I don't think there's a rush to make this work, since it wasn't working before either, so I'm removing the milestone.

**Comment by straight-shoota** [comment]
Okay, just to confirm: The OP code example has the same result on both 1.5.0 and 1.6.0.

**Comment by straight-shoota** [comment]
Yeah, or both actually. We should also write specs for unwanted behaviour, just to document it.
